app/models/data_csv.rb

```diff
@@ -453,15 +429,37 @@ def self.update_with_accreditation
     query_str += 'AND accreditations.accreditation_status IS NOT NULL '
     query_str += "AND LOWER(accreditations.csv_accreditation_type) = 'institutional'; "
 
-    query_str += 'UPDATE data_csvs SET '
-    query_str += 'caution_flag_reason = CONCAT(data_csvs.caution_flag_reason,'
-    query_str += "'Accreditation (', accreditations.accreditation_status, '), ')"
-    query_str += ' FROM accreditations '
-    query_str += 'WHERE data_csvs.cross = accreditations.cross '
-    query_str += 'AND accreditations.cross IS NOT NULL '
-    query_str += %(AND LOWER(accreditations.periods) LIKE '%current%' )
-    query_str += 'AND accreditations.accreditation_status IS NOT NULL '
-    query_str += "AND LOWER(accreditations.csv_accreditation_type) = 'institutional'; "
+    query_str += 'UPDATE data_csvs SET caution_flag_reason = '
```

Comment:
If it helps for review, the basic structure (for all changes today) is a bulk update with 2 nested subqueries. The inner-inner subquery is responsible for getting a distinct set of caution flag reasons. The outer-inner subquery is responsible for aggregating all the cautions for a given key together. The bulk query then concat's the new caution flags to the existing caution flag.

Hopefully this solves the problem of having multiple row caution flags in some csvs and single row caution flags in others.

I will add comments to the next PR.